vengine/vengine/components/UIArea.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+struct UIArea // 2D only
+{
+	glm::ivec2 position;
```

Comment:
If both position and dimension is within the interal resolution translated size, then please make these glm::vec2 rather than glm::ivec2 to more accurately support higher resolution displays.